src/runtime/internal/dev.ts

```diff
@@ -301,6 +301,48 @@ export type ComponentProps<Component extends SvelteComponent> = Component extend
 	? Props
 	: never;
 
+/**
+ * Convenience type to get the events the given component expects. Example:
+ * ```html
+ * <script lang="ts">
+ *    import type { ComponentEvents } from 'svelte';
+ *    import Component from './Component.svelte';
+ *
+ *    type $$Events = ComponentEvent<Component>;
+ * </script>
+ * 
+ * <div class="wrapper">
+ *    <Component on:close />
+ * </div>
+ * ```
+ */
+export type ComponentEvents<Component extends SvelteComponent> = Component extends SvelteComponentTyped<any, infer Events>
+	? Events
+	: never;
+
+/**
+* Convenience type to get an event the given component expects. Example:
+* ```html
+* <script lang="ts">
+*    import type { ComponentEvent } from 'svelte';
+*    import Component from './Component.svelte';
+*
+*    function handleCloseEvent(event: ComponentEvents<Component, 'close'>) {
+*       console.log(event.detail);
+*    };
+* </script>
+*
+* <Component on:close={handleCloseEvent} />
+* ```
+*/
+export type ComponentEvent<Component extends SvelteComponent, Name extends string> = Component extends SvelteComponentTyped<any, infer Events>
+	? never extends Events[Name] ? CustomEvent<unknown> : Events[Name]
+	: never;
+
+export type ComponentSlots<Component extends SvelteComponent> = Component extends SvelteComponentTyped<any, any, infer Slots>
```

Comment:
Since I don't know a use case for this, either, let's leave it until someone comes up with a sensible one